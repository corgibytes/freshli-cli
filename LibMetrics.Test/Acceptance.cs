using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using LibMetrics.Languages.Php;
using LibMetrics.Languages.Ruby;
using Xunit;

namespace LibMetrics.Test
{
  [UseReporter(typeof(DiffReporter))]
  public class Acceptance
  {
    [Fact]
    public void RubyGemsWithGitHistory()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var rubyFixturePath = Fixtures.Path("ruby", "nokotest");
      var results = runner.Run(
        rubyFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsWithHistoryViaGitHub() {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var repoUrl =
        "https://github.com/corgibytes/libmetrics-fixture-ruby-nokotest";
      var results = runner.Run(repoUrl, asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsFeedbinHistoryViaGitHub()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var repoUrl = "https://github.com/feedbin/feedbin";
      var results = runner.Run(repoUrl, asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void ComposerWithoutGitHistory()
    {
      ManifestFinder.Register<PhpComposerManifestFinder>();

      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "large");
      var results = runner.Run(
        phpFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void DrupalComposerWithoutGitHistory()
    {
      ManifestFinder.Register<PhpComposerManifestFinder>();

      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "drupal");
      var results = runner.Run(
        phpFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    }
  }
}
