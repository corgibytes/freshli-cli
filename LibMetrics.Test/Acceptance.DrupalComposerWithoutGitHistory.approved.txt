results[0] = { Date: 1/1/2020 12:00:00 AM, LibYear: { _packagesValues: [ { Name: "drupal/composer_autoloader", Version: "1.2.0", PublishedAt: 3/11/2018 12:00:00 AM, Value: 3.213698630136986 }, { Name: "drupal/drupal", Version: "8.7.11", PublishedAt: 12/18/2019 12:00:00 AM, Value: -0.42191780821917807 },  ], Total: 2.791780821917808 } }
